### PR TITLE
Add ui-react-notifications package

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes
Add new `ui-react-notifications` package for components like InAppMessaging. They will be moved out from `ui-react` in a later breaking change release.

PRs:
- https://github.com/aws-amplify/amplify-ui/pull/3780
- https://github.com/aws-amplify/amplify-ui/pull/3799
- https://github.com/aws-amplify/amplify-ui/pull/3813
- https://github.com/aws-amplify/amplify-ui/pull/3835

#### Description of how you validated changes
POC PR:
- https://github.com/aws-amplify/amplify-ui/pull/3828/

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] Relevant documentation is changed or added (and PR referenced)
- [x] `yarn test` passes and tests are updated/added
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-notifications/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+module.exports = {
+  collectCoverage: true,
+  collectCoverageFrom: ['<rootDir>/src/**/*.(ts|tsx)'],
+  // ignore coverage for top level "export", version and style files
+  coveragePathIgnorePatterns: ['<rootDir>/src/(index|styles|version).(ts|tsx)'],
```
Comment (calebpollman) [inline packages/react-notifications/jest.config.js:5]: There is no version file or _.tsx_ files that we need to ignore right?

Comment (sreeramsama) [reply packages/react-notifications/jest.config.js:5]: yes, will fix it now

Comment (calebpollman) [reply packages/react-notifications/jest.config.js]: Pretty sure i left this feedback on an earlier PR as well. Is there any potential that other changes got dropped?

Comment (sreeramsama) [reply packages/react-notifications/jest.config.js]: That feedback was on Geo PR.
Forgot to make that change here as well.

**Review by reesscot — APPROVED** [review]
LGTM

**Review by calebpollman — APPROVED** [review]
🚢

**Pull request merged**